### DISCUSSION TITLE
Welcome to Gemini-API Discussions!

### DISCUSSION THREAD

**Discussion by dsdanielpark** [dsdanielpark/Gemini-API#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by OlegRuban-ai** [comment]
Hello! 

The idea of ​​parsing is good, I used Bard-Api before. But there is an essential problem, the solution of which will help increase the audience significantly. This is the lack of automatic updating of cookies without the presence of a GUI so that it can be used on the server. At the moment, the official Gemini API is performing well on the servers, but there is no Internet access and the version is Gemini Pro 1.0, and not Gemini Ultra 1.0 as on the site.

**Reply by dsdanielpark** [reply]
Hello! First, thank you for loving both the Bard API and Gemini API.

We're actively exploring ways to automatically update cookies. Development was initially delayed as we prioritized addressing this issue. We attempted auto-updates using browser_cookie3, but it seems that even when cookies are successfully collected, Google's server detects and invalidates them through RotateCookie.

Additionally, development is being delayed due to the need to adjust various requests and parameters, along with rate limiting and temporary account blocks.

Our immediate priority is to port the valuable contributions made to the Bard API and to maintain full functionality of both the Bard and Gemini APIs.

We are eager to develop an alternative method to browser_cookie3 for automatic updates on the server as soon as possible, but finding a workaround has been time-consuming without a clean solution yet.

Currently, cookies extracted from accounts set to Ukraine in the Gemini API remain valid for over 24 hours, but further testing is needed to check for rate limiting.

Further exploration, including Gemini Ultra, is necessary, but this requires extensive experimentation. Reverse engineering means development takes some time.

I'm looking for contributors to help! Contributions are always welcome, and if you're interested in becoming a core maintainer or author, your active participation would be much appreciated.

Thank you once again for your feedback. Have a great day!

**Reply by OlegRuban-ai** [reply]
There are doubts about Ukraine. The fact is that I am from Ukraine and this is indicated in my account settings. But __Secure-1PSIDTS cookies are updated approximately every 30 minutes. I also tried running it on a server with an American IP - the same thing.

**Reply by johnd0e** [reply]
@OlegRuban-ai 

Do you set only this one cookie?
That might be a reason for such short cookie lifetime.

**Reply by dsdanielpark** [reply]
Try with various Google accounts until you find a working cookie. Make sure to log in with a new account after completely resetting the browser to eliminate any remaining cookie values.

Use an incognito window and ensure cookies are used independently. Even if a cookie works, it may not work from different IPs or servers. Providing the entire set of cookies seems to fix one cookie per account. Additionally, once successfully connected with that cookie, it seems to work flawlessly for over three weeks without any errors. Experiment in various environments. Naturally, results may vary depending on country, region, and the state of Google accounts.

**Comment by tejas0111** [comment]
hii !!
Can anyone provide me gpt unofficial api

**Reply by dsdanielpark** [reply]
Create a new Google account and remove all logged-in IDs from your browser. Cookies may not refresh due to synced accounts.

Through the simple video I posted, thoroughly cleanse your browser (log out and delete all accounts, especially synced ones, then completely clear cookies) and try cookies from a different account.

Once it works, your cookies will be valid for a long time (over 3 weeks in my case).

However, I couldn't reuse the valid cookies on another laptop. It seems to be affected by the server or IP as well.
